api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -767,15 +767,34 @@ public CohortDefinition getPatientsWithNextConsultationScheduled175To190Days() {
 
     String mappings = "onOrBefore=${endDate},location=${location}";
     cd.addSearch("TxCurr", EptsReportUtils.map(txCurr, mappings));
-    cd.addSearch("scheduledN3", EptsReportUtils.map(patientsScheduled, mappings));
+    cd.addSearch("scheduledN2", EptsReportUtils.map(patientsScheduled, mappings));
 
     String rapidFlowMappings =
         "onOrAfter=${startDate},onOrBefore=${endDate},locationList=${location}";
     cd.addSearch("rapidFlow", EptsReportUtils.map(rapidFlow, rapidFlowMappings));
     cd.addSearch(
         "completed", EptsReportUtils.map(getPatientsWhoCompletedRapidFlow(), rapidFlowMappings));
 
-    cd.setCompositionString("TxCurr AND (scheduledN3 OR rapidFlow) NOT completed");
+    cd.addSearch(
+        "pregnant",
+        EptsReportUtils.map(
+            txNewCohortQueries.getPatientsPregnantEnrolledOnART(),
+            "startDate=${endDate-9m},endDate=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "breastfeeding",
+        EptsReportUtils.map(
+            getBreastfeedingComposition(),
+            "onOrAfter=${endDate-18m},onOrBefore=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "tbPatient",
+        EptsReportUtils.map(
+            commonCohortQueries.getPatientsOnTbTreatment(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString(
+        "TxCurr AND (scheduledN2 OR rapidFlow) NOT (completed OR pregnant OR breastfeeding OR tbPatient)");
```

Comment:
@JoaoCumbula please change this block of pregnant, breastfeeding and TB to 
```suggestion
        "TxCurr AND (scheduledN2 OR rapidFlow) NOT (completed OR PregnantAndBreastfeedingAndOnTBTreatment)");
```

Reply:
@cfaife Already did this change and fix the compile issues, all i did is fix the mapping on DataSet class. Pull request already sent.